Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -572,6 +595,13 @@ public struct Decimal128: BSONNumber, Equatable, Codable, CustomStringConvertibl
      }
 }
 
+// An extension of `Decimal128` to add capability to be hashed
+extension Decimal128: Hashable {
```

Comment:
[nit] This is my own fault for giving you potentially bad advice, but I think these should actually be moved inwards into the definitions themselves instead of in extensions. I only say that because it seems that everywhere else in the file, that's the way things are done.

What do you think, @kmahar?

Reply:
I think I'm pro extension actually. while this doesn't quite line up with the rest of this file I think it does line up with the trend we've followed elsewhere in the driver.

I think at some point in the future (probably when I get around to splitting up this massive file into separate files by type) the other conformances should be moved out to extensions, too.

`Codable` conformance often can't be in an extension because of rules about required initializers, but we can put conformance for stuff like `Equatable` and `CustomStringConvertible` in extensions. 

I like having it in the extensions because it makes it very clear what parts of the API are relevant to which protocol.

Reply:
Okay, my original comment can be disregarded then.